magento/magento2
# Add missing entity type constraint to catalog URL rewrite check

### Description (*)


This fixes the error
```
main.CRITICAL: Notice: Trying to access array offset on value of type bool in vendor/magento/module-catalog-url-rewrite/Model/Storage/DynamicStorage.php on line 182 
```

Which happens if one tries to access a URL which ends with a category URL rewrite. It happens because `findProductRewriteByRequestPath` in that class doesn't constrain the `$productFromDb` select with the `product` entity type (or `$categoryFromDb` with the `category` entity type for that matter, but it doesn't directly contribute to this bug), and falsely returns the category URL rewrite for `$productFromDb`. This causes the method to continue even though it should return null at this point, leading to it trying to access the `redirect_type` property of the falsy `$categoryFromDb` on line 182, which generates the notice and causes the error report to show.

### Manual testing scenarios (*)

1. Create a category with the name "Test Category" and set the URL path as "test-category". Doesn't matter at which level the category is.
2. Access "<base_url>/asdf/test-category". The "asdf" here is just any fake path in the store, meaning that normally a 404 page should show.

Due to this bug a notice will be shown if you're in developer mode, a Magento error report otherwise.

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#31225: Add missing entity type constraint to catalog URL rewrite check

### Resolved issues:
1. [ ] resolves magento/magento2#31235: Add missing entity type constraint to catalog URL rewrite check
